prover/docs/01_gcp_vm.md

```diff
@@ -0,0 +1,140 @@
+
+## Adding your own ssh key
+
+Using browser to connect to the machine may not be the most convenient option. Instead, we can add an SSH key to be able
+to connect there.
+
+It is highly recommended to generate a new SSH key specifically for this VM, for example:
+
+```
+ssh-keygen -t rsa -f ~/.ssh/gcp_vm -C <YOUR WORK EMAIL> -b 2048
+```
+
+...where "your work email" is the same email you use to access GCP.
+
+Check the contents of the public key:
+
+```
+cat ~/.ssh/gcp_vm.pub
+```
+
+Click on your machine name, then click on "Edit". Scroll down until you see "SSH Keys" section and add the generated
+public key there. Then save.
+
+Get back to the list of VMs and find the external IP of your VM. Now you should be able to connect to the VM via ssh.
+Assuming that your work email is `abc@example.com` and the external IP is 35.35.35.35:
+
+```
+ssh -i ~/.ssh/gcp_vm abc@35.35.35.35
+```
+
+## Make the VM cozy
+
+If you intend to use the VM somewhat regularly, install all the tools you would normally install on your own machine,
+like `zsh` and `nvim`.
+
+It is also _highly recommended_ to install `tmux`, as you will have to run multiple binaries and observe their output.
+If you don't know what is it or why should you care, watch [this video](https://www.youtube.com/watch?v=DzNmUNvnB04).
+
+Native `tmux` may be hard to use, so you may also want to install some configuration for it, e.g.
+
+- [oh-my-tmux](https://github.com/gpakosz/.tmux) or
+- [tmux-sensible](https://github.com/tmux-plugins/tmux-sensible).
+
+Finally, it is recommended to choose a different terminal theme than what you use locally, so that you can easily see
+whether you're running in the VM or locally.
+
+## Connecting via VS Code
+
+VS Code can connect to VMs via SSH, so you can have the comfort of using your own IDE while still running everything on
+a remote machine.
+
+If you're using WSL, note that VS Code will have to look up the keys in Windows, so you will have to copy your keys
+there as well, e.g.:
+
+```
+cp ~/.ssh/gcp_vm* /mnt/c/Users/User/.ssh
+```
+
+Then, when you open a fresh VS Code window, in the "Start" section:
+
+- Choose "Connect to Host"
+- Click on "Configure Hosts"
+- Create a host entry.
+
+Host entry looks as follows:
+
+```
+Host <host_name>
+  HostName <external IP>
+  IdentityFile <path to private SSH key>
+  User <your user name in VM>
+```
+
+E.g. for the command we've used as an example before: `ssh -i ~/.ssh/gcp_vm abc@35.35.35.35`, the file will be:
+
+```
+Host gcp_vm
+  HostName 35.35.35.35
+  IdentityFile ~/.ssh/gcp_vm
+  User abc
+```
+
+Once you've configured the host, you can click on "Connect to" again, then "Connect to Host", and your VM should be
+listed there. On the first connect you'll have to confirm that you want to connect to it, and then choose the operating
+system (Linux).
+
+## Generating the SSH key on VM
+
+Last but not least, you probably want to make sure that you can access GitHub from your VM, so that you can push and
```

Comment:
this is risky BTW.. as any other owner of the GCP project (so everyone in matterlabs) can see those SSH keys.

So you might want to restrict them only to a given repo etc.